rosdoc2/verbs/build/builders/sphinx_builder.py

```diff
@@ -615,6 +617,28 @@ def build(self, *, doc_build_folder, output_staging_directory):
             if package_name != self.build_context.package.name
         ]
 
+        # Collect package-only exec_depends
+        exec_depends = self.build_context.package.exec_depends
+        ros_distro = os.environ.get('ROS_DISTRO')
+        if not ros_distro:
+            logger.warning('ROS_DISTRO not set, cannot check ros package dependencies')
```

Comment:
Historically we've wanted to make sure that we weren't locking in too closely onto the ROS_DISTRO. But adding it with this backwards compatibility seems like a good compromise.